src/Configuration.php

```diff
@@ -0,0 +1,347 @@
+                'description' => 'Override the default ProcessManager implementation with another class (unstable)',
+                'default' => ProcessManager::class,
+            ],
+        ];
+    }
+
+    /**
+     * Returns the default configuration options table
+     *
+     * @return array
+     */
+    public static function getDefaults()
+    {
+        return array_map(function ($mapping) {
+            return $mapping['default'];
+        }, self::getMapping());
+    }
+
+    /**
+     * Load a config from a file path
+     *
+     * @param string $path
+     * @return Configuration
+     */
+    public static function loadFromPath($path)
+    {
+        $content = file_get_contents($path);
+        $config = json_decode($content, true);
+        return new Configuration($config);
+    }
+
+    /**
+     * Configuration constructor.
+     *
+     * @param array $options
+     * @param array $arguments
+     */
+    public function __construct($options = [], $arguments = [])
+    {
+        $this->options = array_merge(self::getDefaults(), $options);
+        $this->arguments = $arguments;
+    }
+
+    /**
+     * Attempt to resolve a cgi-path value automatically.
+     *
+     * @return null|string
+     */
+    public function resolvePhpCgiPath()
+    {
+        $executableFinder = new PhpExecutableFinder();
+        $binary = $executableFinder->find();
+
+        $cgiPaths = [
+            $binary . '-cgi', //php7.0 -> php7.0-cgi
+            str_replace('php', 'php-cgi', $binary), //php7.0 => php-cgi7.0
+        ];
+
+        foreach ($cgiPaths as $cgiPath) {
+            $path = trim(`which $cgiPath`);
+            if ($path) {
+                return $path;
+            }
+        }
+
+        return null;
+    }
+
+    public function tryResolvePhpCgiPath()
+    {
+        $this->arguments['cgi-path'] = $this->resolvePhpCgiPath();
+    }
+
+    /**
+     * Set the arguments array
+     *
+     * @param array $arguments
+     */
+    public function setArguments($arguments)
+    {
+        $this->arguments = $arguments;
+    }
+
+    /**
+     * Get the arguments array
+     *
+     * @return array
+     */
+    public function getArguments()
+    {
+        return $this->arguments;
+    }
+
+    /**
+     * @param array $option
+     * @return mixed
+     */
+    public function getOption($option)
+    {
+        return isset($this->arguments[$option]) ? $this->arguments[$option] : $this->options[$option];
+    }
+
+    /**
+     * @return string
+     */
+    public function getBridge()
+    {
+        return $this->getOption('bridge');
+    }
+
+    /**
+     * @return string
+     */
+    public function getAppEnv()
+    {
+        return $this->getOption('app-env');
+    }
+
+    /**
+     * @return string
+     */
+    public function getAppBootstrap()
+    {
+        return $this->getOption('bootstrap');
+    }
+
+    /**
+     * @return bool
+     */
+    public function isPopulateServer()
```

Comment:
Shall we change all these is... to get...?

Reply:
Naming conventions for booleans often use `is` for getters. This is the case for Java, at least. It comes down to what is more readable.

Reply:
ok